index.js

```diff
@@ -1460,13 +1460,74 @@ module.exports = function (log, indexesPath) {
     )
   }
 
+  function reindex(offset, cb) {
+    let seq = 0
+    if (offset === 0) {
+      seq = 0
+      offset = -1
+    } else if (offset !== -1) {
+      const { tarr } = indexes['seq']
+      let prevOffset = 0
+      for (const len = tarr.length; seq < len; ++seq)
+        if (tarr[seq] === offset) {
+          offset = prevOffset
+          break
+        } else prevOffset = tarr[seq]
+    }
+
+    function resetIndex(index) {
+      if (index.offset >= offset) {
+        if (index.count) index.count = seq
+
+        if (index.map) {
+          // prefix map pushes to arrays, so we need to clean up
+          for (let [prefix, arr] of Object.entries(index.map)) {
+            index.map[prefix] = arr.filter((x) => x < seq)
+          }
+        }
+
+        index.offset = offset
+      }
+    }
+
+    push(
+      push.values(Object.entries(indexes)),
+      push.asyncMap(([indexName, index], cb) => {
```

Comment:
Neat use of `values(Object.entries` and `asyncMap`. Oh, and it's push-stream. I have a vague memory that push-stream `asyncMap` was incomplete or unmaintained. Tread carefully